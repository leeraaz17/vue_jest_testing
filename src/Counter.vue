<template>
<div>
    <p>Count: {{ counter }}</p>
    <button class="btn btn-info" @click="increased()">Increment</button>
</div>
</template>

<script>
export default {
    data() {
        return {
            counter: 0
        }
    },
    methods: {
        increased() {
            this.counter++;
        }
    }
}
</script>