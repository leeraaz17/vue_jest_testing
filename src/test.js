import { shallowMount } from '@vue/test-utils'
import App from './App'
import Counter from './Counter'

describe('App', () => {
    const wrapper = shallowMount(App)  

    test('is a Vue instance', () => {
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('contains the increment button', () => {
        const wrapper = shallowMount(Counter)
        expect(wrapper.vm.counter).toBe(0)
        wrapper.find('button').trigger('click')
        expect(wrapper.vm.counter).toBe(1)
    })

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
})