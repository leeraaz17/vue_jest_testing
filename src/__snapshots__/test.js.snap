// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App contains the increment button 1`] = `"<div id=\\"app\\"><counter-stub></counter-stub></div>"`;

exports[`App renders correctly 1`] = `
<div
  id="app"
>
  <counter-stub />
</div>
`;
